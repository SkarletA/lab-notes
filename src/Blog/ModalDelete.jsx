/* eslint-disable react/prop-types */
/* eslint-disable import/prefer-default-export */
import React from 'react';
import style from './ModalDelete.module.css';
import {
  deleteDoc,
  doc,
} from '../Firebase/firebase';
import { db } from '../Firebase/firebaseconfig';

export function ModalDelete({ openModalDelete, closeModalDelete, currentId }) {
  const onDeleteBlog = async (id) => {
    await deleteDoc(doc(db, 'blogs', id));
  };
  console.log(currentId);

  if (!openModalDelete) return null;
  return (
    <section className={style.modalDelete}>
      <div className={style.popupDelete}>
        <p className={style.titleDelete}>Are you sure want to delete this note? </p>
        <div className={style.btnsDelete}>
          <button
            type="button"
            className={style.btnDelete}
            onClick={() => {
              onDeleteBlog(currentId);
              closeModalDelete();
            }}
          >
            Delete
          </button>
          <button
            type="button"
            className={style.btnNoDelete}
            onClick={closeModalDelete}
          >
            Cancel
          </button>
        </div>
      </div>
    </section>
  );
}
