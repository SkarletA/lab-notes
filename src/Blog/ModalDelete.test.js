/* eslint-disable no-shadow */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { ModalDelete } from './ModalDelete';

describe(('Test component ModalDelete'), () => {
  test('render component ModalDelete', () => {
    let modalDelete;
    act(() => {
      modalDelete = create(
        <Router>
          <ModalDelete />
        </Router>,
      );
    });
    expect(modalDelete.toJSON()).toMatchSnapshot();
  });
});
