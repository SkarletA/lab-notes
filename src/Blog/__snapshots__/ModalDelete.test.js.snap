// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component ModalDelete render component ModalDelete 1`] = `null`;
